src/Kernel/client/plotting.ts

```diff
@@ -65,23 +65,25 @@ export function updateChart(plotStyle: PlotStyle, chart: ChartJs, state: Display
 
 function updateWithAmplitudePhaseData(chart: ChartJs, state: DisplayableState) {
     let amps = state.amplitudes;
-    let newCount = amps.length;
+    let nBasisStates = amps.length;
+    let nBitLength = Math.ceil(Math.log2(nBasisStates));
+
     chart.data = {
-        labels: Array.from(Array(newCount).keys()).map(idx => {
-            let bitstring = (idx >>> 0).toString(2).padStart(newCount, "0");
+        labels: Array.from(Array(nBasisStates).keys()).map(idx => {
+            let bitstring = (idx >>> 0).toString(2).padStart(nBitLength, "0");
             return `|${bitstring}⟩`;
         }), //basis state labels
```

Comment:
```suggestion
        labels: Array.from(Array(nBasisStates), (_, idx) => {
            let bitstring = (idx >>> 0).toString(2).padStart(nBitLength, "0");
            return `|${bitstring}⟩`;
        }), //basis state labels
```

Same for the ones below.

Reply:
It looks like there's quite a few places that uses the same logic as this and requires the same change. I commented on a few of these but left some out. Is there a way to write this as a helper function instead of duplicating this logic?

Reply:
Agreed, it'd be good to isolate basis state labeling into its own function, so that the labeling logic can be modified based on configuration options and so forth as well.